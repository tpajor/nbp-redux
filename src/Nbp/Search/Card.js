import React from 'react';
import PropTypes from 'prop-types';

import './Card.css';

const Card = ({ cardData, isTemporary, deleteCard, userSignedIn, signedIn, addCard, showDetails, cards }) => {
  return (
    <div className={'Card'}>
      <div className={'row CardMainContainer'}>
        <div className={'col-4 CardMainCell Left'}>
          <p className={'CardMainCellText'}>
            {cardData.code}
          </p>
        </div>
        <div className={'col-4 CardMainCell Mid'}>
          <p className={'CardMainCellText'}>
            {cardData.rates[0].ask}  
          </p>  
        </div>
        <div className={'col-4 CardMainCell Right'}>
          <p className={'CardMainCellText'}>
            {cardData.rates[0].bid}
          </p>
          {isTemporary ? 
            '' :
            <button className={'DeleteCardButton'} onClick={() => deleteCard(cardData.id, userSignedIn, signedIn)}>X</button>
          }
        </div>
      </div>
      <div className={'CardNameCell'}>
        <p>
          {cardData.currency}
        </p>
      </div>
      { (isTemporary) ?
        <div className={'DoubleButton'}>
          <button className={'SaveButton'} onClick={() => addCard(cardData, userSignedIn, signedIn, cards)}>
            Przypnij
          </button> 
          <button className={'HalfCardButton'} onClick={() => showDetails(cardData.code)}>
            Dodaj do wykresu
          </button>
        </div> :
        <button className={'CardButton'} onClick={() => showDetails(cardData.code)}>
          Dodaj do wykresu
        </button>
      }
    </div>
  );
};

Card.propTypes = {
  userSignedIn: PropTypes.string,
  signedIn: PropTypes.bool,
  cardData: PropTypes.object,
  addCard: PropTypes.func,
  showDetails: PropTypes.func,
  isTemporary: PropTypes.bool,
  deleteCard: PropTypes.func,
  cards: PropTypes.array,
};

export default Card;