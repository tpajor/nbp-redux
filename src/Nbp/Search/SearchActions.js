import axios from 'axios';

import getUniqueId from '../../Services/uniqueIdGenerator';
import callApi from '../../Services/apiCaller';
import { fetchUsers } from '../../Login/LoginActions';

export const ADD_CARD = 'ADD_CARD';
export const ADD_CARD_ERROR = 'ADD_CARD_ERROR';
export const DELETE_CARD = 'DELETE_CARD';
export const SHOW_DETAILS = 'SHOW_DETAILS';
export const GET_CURRENCY = 'GET_CURRENCY';
export const SEARCH_INPUT_ERROR = 'SEARCH_INPUT_ERROR';
export const GET_CURRENCIES_TABLE = 'GET_CURRENCIES_TABLE';
export const POPULATE_CARDS = 'POPULATE_CARDS';

export function addCard(card) {
  return {
    type: ADD_CARD,
    card,
  };
}

export function addCardError() {
  return {
    type: ADD_CARD_ERROR,
  }
}

export function addCardRequest(card, userName, isSignedIn, cards) {
  return (dispatch) => {
    const cardExists = cards.find(cardFromCardsArray => cardFromCardsArray.code === card.code);
    if (cardExists) {
      return dispatch(addCardError());
    } else {
      const id = getUniqueId();
      if (isSignedIn) {
        return callApi('card', 'post', { card: { ...card, id }, userName }).then(res => {
          dispatch(addCard({ ...card, id }));
          dispatch(fetchUsers());
        });
      } else {
        return dispatch(addCard({ ...card, id }));
      }
      }
  };
}

export function deleteCard(cardId) {
  return {
    type: DELETE_CARD,
    cardId: cardId,
  };
}

export function deleteCardRequest(cardId, userName, isSignedIn) {
  return (dispatch) => {
    if (isSignedIn) {
      return callApi('card', 'delete', { cardId, userName }).then(res => {
        dispatch(deleteCard(cardId));
        dispatch(fetchUsers());
      });
    } else {
      dispatch(deleteCard(cardId));
    }
  }
}

export function showDetails(currencyCode) {
  return {
    type: SHOW_DETAILS,
    currencyCodeToViewInDetail: currencyCode,
  };
}

export function getCurrency(card) {
  return {
    type: GET_CURRENCY,
    temporaryCard: card,
  };
}

export function searchInputError(searchErrorMessage) {
  return {
    type: SEARCH_INPUT_ERROR,
  };
}

export function getCurrencyRequest(currencyCode) {
  return (dispatch) => {
    return axios.get(`https://api.nbp.pl/api/exchangerates/rates/c/${currencyCode}/today/`)
      .then(res => {
        dispatch(getCurrency(res.data));
      })
      .catch(() => {
        dispatch(searchInputError());
      });
  }
}

export function getCurrenciesTable(table) {
  return {
    type: GET_CURRENCIES_TABLE,
    table,
  };
}

export function getCurrenciesTableRequest() {
  return (dispatch) => {
    return axios.get(`https://api.nbp.pl/api/exchangerates/tables/c/today/`)
     .then(res => {
       dispatch(getCurrenciesTable(res.data));
     });
  }
}

export function populateCards(cards) {
  return {
    type: POPULATE_CARDS,
    cards
  };
}