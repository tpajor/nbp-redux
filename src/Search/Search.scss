.Form {
  margin: 20px 0 20px 0;
  display: inline-block;
  vertical-align: middle;
  width: 100%;
  position: relative;
}

.SearchButton {
  position: absolute;
  margin-top: 20px;
}
