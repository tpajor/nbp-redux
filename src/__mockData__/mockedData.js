const card = { id: 'someId', code: 'USD', currency: 'dolar amerykański', rates: [{ ask: 3.3, bid: 3.4 }]};
const card2 = { id: 'someId2', code: 'USD', currency: 'dolar amerykański', rates: { ask: 3.3, bid: 3.4 }};

const search = {
  cards: [card, card2],
  temporaryCard: card,
  searchError: false,
  addCardError: true,
  addCardErrorMessage: 'error',
  currenciesTable: ['usd', 'chf', 'aud'],
  currencyCodeToViewInDetail: 'usd',
};

const user = { userName: 'User', pass: 'pass' };
const user2 = { userName: 'User2', pass: 'pass2'};

export const login = {
  signingError: true,
  signingErrorMessage: 'error',
  signedIn: true,
  userSignedIn: 'User',
  users: [user, user2]
};

const lastRates = [
  { ask: 3.5983, bid: 3.5271, effectiveDate: "2018-05-07", no: "087/C/NBP/2018" },
  { ask: 3.6088, bid: 3.5374, effectiveDate: "2018-05-08", no: "088/C/NBP/2018" }
];

const detail = {
  lastRates: lastRates,
  detailsInputError: false,
  detailsInputErrorMessage: '',
};

export const initialState = { search, login, detail };

export const searchViewData = {
  cards: search.cards,
  temporaryCard: search.temporaryCard,
  searchError: search.searchError,
  signedIn: login.signedIn,
  userSignedIn: login.userSignedIn,
  addCardError: search.addCardError,
  addCardErrorMessage: search.addCardErrorMessage,
  currenciesTable: search.currenciesTable,
};

export const detailViewData = {
  currencyCodeToViewInDetail: search.currencyCodeToViewInDetail,
  lastRates: detail.lastRates,
  detailsInputError: detail.detailsInputError,
  detailsInputErrorMessage: detail.detailsInputErrorMessage,
  signedIn: login.signedIn,
};