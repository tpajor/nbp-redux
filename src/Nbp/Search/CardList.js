import React from 'react';
import Card from './Card';
import PropTypes from 'prop-types';

const CardList = ({ cards, signedIn, userSignedIn, deleteCard, showDetails }) => {
  return (
    <div>
      {cards.map(card => 
        <Card key={card.id}
          signedIn={signedIn}
          userSignedIn={userSignedIn}
          deleteCard={deleteCard}
          cardData={card}
          isTemporary={false}
          showDetails={showDetails}
          cards={cards}
        />
      ).reverse()}
    </div>
  );
};

CardList.propTypes = {
  cards: PropTypes.array,
  deleteCard: PropTypes.func,
  showDetails: PropTypes.func,
  signedIn: PropTypes.bool,
  userSignedIn: PropTypes.string,
};

export default CardList;