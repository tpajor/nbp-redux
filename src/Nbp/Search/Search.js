import React from 'react';
import PropTypes from 'prop-types';
import { throttle } from "throttle-debounce";

import './Search.css';

export default class Search extends React.Component {
  constructor(props) {
    super(props);
    this.state = { 
      currencyCode: '',
      autocomplete: [],
      inputError: false,
      activeItemPosition: 0,
      hover: false,
    };
    this.setAutocompleteDebounce = throttle(200, this.setAutocomplete);
  }
  

  handleFocus = () => {
    this.props.getCurrenciesTable();
  }

  handleSubmit = event => {
    event.preventDefault();
    const listOfMatchingCurrencies = this.props.currenciesTable.filter(currency => currency === this.state.currencyCode.toUpperCase());
    if (listOfMatchingCurrencies.length === 1) {
      this.setState({ autocomplete: [], inputError: false });
      this.props.getCurrency(this.state.currencyCode.toLowerCase());
    } else {
      this.setState({ inputError: true });
    }
  }

  setAutocomplete = input => {
    const listOfAutocompleteMatchingInput = this.props.currenciesTable.filter(currency => currency.indexOf(input.toUpperCase()) >= 0);
    if (listOfAutocompleteMatchingInput.length > 0) {
      if (input.length > 0) {
        this.setState({ autocomplete: listOfAutocompleteMatchingInput, inputError: false, activeItemPosition: 0 });
      } else {
        this.setState({ autocomplete: [], inputError: false, activeItemPosition: 0 })
      }
    } else {
      this.setState({ autocomplete: [], inputError: true, activeItemPosition: 0 });
    }
  }

  handleChange = event => {
    this.setState({ currencyCode: event.target.value }, () => {
      this.setAutocompleteDebounce(this.state.currencyCode);
    });
  }

  handleKeyDown = event => {
    if (event.keyCode === 38 && this.state.activeItemPosition > 0) {
      this.setState(prevState => ({ activeItemPosition: prevState.activeItemPosition - 1 }));
    } else if (event.keyCode === 40 && this.state.activeItemPosition < this.state.autocomplete.length - 1) {
      this.setState(prevState => ({ activeItemPosition: prevState.activeItemPosition + 1 }));
    } else if (event.keyCode === 13) {
      if (this.state.currencyCode === this.state.autocomplete[0] && this.state.autocomplete.length === 1) {
        this.handleSubmit(event);
      } else if (this.state.autocomplete.length > 0) {
        const chosenCurrency = this.state.autocomplete[this.state.activeItemPosition];
        this.setState({ currencyCode: chosenCurrency, autocomplete: [].concat(chosenCurrency) })
      }
    }
  }

  handleClick = (event, currency) => {
    event.preventDefault();
    this.setState({ currencyCode: currency, autocomplete: this.state.autocomplete.filter(currencyToMatch => currencyToMatch === currency) })
  }

  handleMouseEnter = event => {
    event.preventDefault();
    this.setState({ hover: true });
  }

  handleMouseLeave = event => {
    event.preventDefault();
    this.setState({ hover: false });
  }

  handleHoverChoice = (event, i) => {
    event.preventDefault();
    this.setState({ activeItemPosition: i })
  }

  render() {
    const { searchError } = this.props;
    const { currencyCode, inputError, activeItemPosition, hover } = this.state;
    return (
      <form onSubmit={this.handleSubmit} className={'Form'}>
        <div className={'form-group col-8 Form FormInput'}>
          <input className={`form-control  ${(inputError || searchError) ? 'is-invalid' : ''}`}
            onFocus={this.handleFocus}
            type="text" 
            value={currencyCode}
            onBlur={this.handleBlur}
            onChange={this.handleChange}
            onKeyDown={this.handleKeyDown}
            placeholder="Wpisz kod waluty"
          />
          {(this.state.autocomplete.length > 0) ? 
            <ol className="CurrencyList" onMouseEnter={this.handleMouseEnter} onMouseLeave={this.handleMouseLeave}>
              {this.state.autocomplete.map((currency, i) => <li key={i} className={'Currency'} onMouseEnter={event => this.handleHoverChoice(event, i)}>
                <button onClick={event => this.handleClick(event, currency)} className={`CurrencyButton ${(activeItemPosition === i && !hover) ? 'active' : ''}`}>
                  <p className="CurrencyText">{currency}</p>
                </button>
              </li>)}
            </ol>: 
            ''
          }
          {(inputError) ?
            <p style={{ color: 'red' }}>Wpisz trzyliterowy kod waluty, np. "USD", "chf"</p> :
            ''
          }
          {(searchError) ?
            <p style={{ color: 'red' }}>Błąd serwera. Spróbuj ponownie później.</p> :
            ''
          }
        </div>
        <button type="submit" className={'btn btn-primary col-4 SearchButton'}>Szukaj</button>
      </form>
    );
  }
};

Search.propTypes = {
  getCurrency: PropTypes.func,
  searchError: PropTypes.bool,
  getCurrenciesTable: PropTypes.func,
  currenciesTable: PropTypes.array,
};