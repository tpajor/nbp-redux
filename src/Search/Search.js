import React from 'react';

import './Search.css';

export default class Search extends React.Component {
  state = { 
    currencyCode: '',
  };

  handleSubmit = event => {
    event.preventDefault();
    this.props.searchByCode(this.state.currencyCode.toLowerCase());
  }
  
  render() {
    return (
      <form onSubmit={this.handleSubmit} className={'Form'}>
        <div className={'form-group col-8 Form'}>
          <input className={`form-control  ${this.props.inputError ? 'is-invalid' : ''}`}
            type="text" 
            value={this.state.currencyCode}
            onChange={event => this.setState({ currencyCode: event.target.value })}
            placeholder="Wpisz kod waluty"
          />
          {this.props.inputError ?
            <p style={{ color: 'red' }}>{this.props.errorMessage}</p> :
            ''
          }
        </div>
        <button type="submit" className={'btn btn-primary col-4 SearchButton'}>Szukaj</button>
      </form>
    );
  }
}