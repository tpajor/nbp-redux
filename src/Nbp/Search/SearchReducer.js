import { ADD_CARD, ADD_CARD_ERROR, DELETE_CARD, SHOW_DETAILS, GET_CURRENCY, SEARCH_INPUT_ERROR, GET_CURRENCIES_TABLE, POPULATE_CARDS } from './SearchActions';

export const initialState = {
  cards: [],
  addCardError: false,
  addCardErrorMessage: '',
  currencyCodeToViewInDetail: '',
  temporaryCard: null,
  searchError: false,
  currenciesTable: [],
}

export default function search(state = initialState, action) {
  switch (action.type) {
    case ADD_CARD: {
      const updateCards = state.cards.concat(action.card);
      return { ...state, cards: updateCards, addCardError: false, addCardErrorMessage: '', };
    }

    case ADD_CARD_ERROR: {
      return { ...state, addCardError: true, addCardErrorMessage: 'Karta tej waluty jest już przypięta' }
    }

    case DELETE_CARD: {
      const updatedCards = state.cards.filter(card => card.id !== action.cardId)
      return { ...state, cards: updatedCards };
    }

    case SHOW_DETAILS: {
      return { ...state, currencyCodeToViewInDetail: action.currencyCodeToViewInDetail };
    }

    case GET_CURRENCY: {
      return { ...state, 
        temporaryCard: action.temporaryCard, 
        searchError: false,
      };
    }

    case SEARCH_INPUT_ERROR: {
      return { ...state, temporaryCard: null, searchError: true }
    }

    case GET_CURRENCIES_TABLE: {
      const newCurrenciesTable = action.table[0].rates.map(currency => currency.code);
      return { ...state, currenciesTable: newCurrenciesTable };
    }

    case POPULATE_CARDS: {
      return { ...initialState, cards: action.cards };
    }

    default: {
      return { ...state };
    }
  }
}