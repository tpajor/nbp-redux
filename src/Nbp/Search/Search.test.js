import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Search from './Search';
import { searchViewData } from '../../__mockData__/mockedData'


Enzyme.configure({ adapter: new Adapter() });

const initialState = { 
  currencyCode: '',
  autocomplete: [],
  inputError: false,
  activeItemPosition: 0,
  hover: false,
};

function setup() {
  const props = {
    searchError: searchViewData.searchError,
    currenciesTable: searchViewData.currenciesTable,
    getCurrency: jest.fn(),
    getCurrenciesTable: jest.fn(),
  };

  const enzymeWrapper = shallow(<Search {...props} />);

  return {
    enzymeWrapper,
    props
  };
};

describe('Card component', () => {
  it('should render self and subcomponents with their props', () => {
    const { enzymeWrapper, props } = setup();

    expect(enzymeWrapper.length).toBe(1);

    expect(enzymeWrapper.find('form').length).toBe(1);
    expect(enzymeWrapper.find('form').hasClass('Form')).toBe(true);

    expect(enzymeWrapper.find('div').length).toBe(1);
    expect(enzymeWrapper.find('div').hasClass('form-group col-8 Form FormInput')).toBe(true);

    expect(enzymeWrapper.find('button').length).toBe(1);
    expect(enzymeWrapper.find('button').hasClass('btn btn-primary col-4 SearchButton')).toBe(true);

    expect(enzymeWrapper.find('input').length).toBe(1);
    expect(enzymeWrapper.find('input').hasClass('form-control ')).toBe(true);
    enzymeWrapper.setProps({ ...props, searchError: true });
    expect(enzymeWrapper.find('input').hasClass('form-control is-invalid')).toBe(true);
    enzymeWrapper.setProps({ ...props });
    enzymeWrapper.setState({ ...initialState, inputError: true });
    expect(enzymeWrapper.find('input').hasClass('form-control is-invalid')).toBe(true);
    enzymeWrapper.setState({ ...initialState });

    expect(enzymeWrapper.find('p').length).toBe(0);
    enzymeWrapper.setProps({ ...props, searchError: true });
    expect(enzymeWrapper.find('p').length).toBe(1);
    expect(enzymeWrapper.find('p').text()).toEqual('Błąd serwera. Spróbuj ponownie później.');
    enzymeWrapper.setState({ ...initialState, inputError: true });
    expect(enzymeWrapper.find('p').length).toBe(2);
    enzymeWrapper.setProps({ ...props });
    expect(enzymeWrapper.find('p').length).toBe(1);
    expect(enzymeWrapper.find('p').text()).toEqual('Wpisz trzyliterowy kod waluty, np. "USD", "chf"');
    enzymeWrapper.setState({ ...initialState });

    expect(enzymeWrapper.find('ol').length).toBe(0);
    enzymeWrapper.setState({ ...initialState, autocomplete: props.currenciesTable });
    expect(enzymeWrapper.find('ol').length).toBe(1);
    expect(enzymeWrapper.find('li').length).toEqual(props.currenciesTable.length);
    expect(enzymeWrapper.find('p').length).toEqual(props.currenciesTable.length);
    expect(enzymeWrapper.find('button').length).toEqual(props.currenciesTable.length + 1);
  });

  it('should render correctly', () => {
    const { enzymeWrapper } = setup();

    expect(enzymeWrapper).toMatchSnapshot();
  });

  it('should get currencies table for autocomplete onFocus on input', () => {
    const { enzymeWrapper, props } = setup();
    const getCurrenciesTableInput = enzymeWrapper.find('input');
    getCurrenciesTableInput.simulate('focus');
    expect(props.getCurrenciesTable.mock.calls.length).toBe(1);
  });

  it('should get currency data from nbp api if the currency exists in api', () => {
    const { enzymeWrapper, props } = setup();
    const getCurrencyForm = enzymeWrapper.find('form');
    getCurrencyForm.simulate('submit', { preventDefault: () => {} });
    expect(props.getCurrency.mock.calls.length).toBe(0);
    expect(enzymeWrapper.state([ 'inputError' ])).toBe(true);

    enzymeWrapper.setState({ ...initialState, currencyCode: 'usd', inputError: true, autocomplete: ['USD'] });
    enzymeWrapper.setProps({ ...props, currenciesTable: props.currenciesTable.map(currency => currency.toUpperCase()) });
    getCurrencyForm.simulate('submit', { preventDefault: () => {} });
    expect(props.getCurrency.mock.calls.length).toBe(1);
    expect(enzymeWrapper.state([ 'inputError' ])).toBe(false);
    expect(enzymeWrapper.state([ 'autocomplete' ])).toEqual([]);
  });
});