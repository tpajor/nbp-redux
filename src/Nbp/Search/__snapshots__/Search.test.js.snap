// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    currenciesTable={
        Array [
            "usd",
            "chf",
            "aud",
          ]
    }
    getCurrenciesTable={[Function]}
    getCurrency={[Function]}
    searchError={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="form-group col-8 Form FormInput"
>
          <input
                    className="form-control "
                    onBlur={undefined}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Wpisz kod waluty"
                    type="text"
                    value=""
          />
          
          
          
</div>,
        <button
          className="btn btn-primary col-4 SearchButton"
          type="submit"
>
          Szukaj
</button>,
      ],
      "className": "Form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="form-control "
              onBlur={undefined}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Wpisz kod waluty"
              type="text"
              value=""
/>,
            "",
            "",
            "",
          ],
          "className": "form-group col-8 Form FormInput",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control ",
              "onBlur": undefined,
              "onChange": [Function],
              "onFocus": [Function],
              "onKeyDown": [Function],
              "placeholder": "Wpisz kod waluty",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "",
          "",
          "",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Szukaj",
          "className": "btn btn-primary col-4 SearchButton",
          "type": "submit",
        },
        "ref": null,
        "rendered": "Szukaj",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="form-group col-8 Form FormInput"
>
            <input
                        className="form-control "
                        onBlur={undefined}
                        onChange={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        placeholder="Wpisz kod waluty"
                        type="text"
                        value=""
            />
            
            
            
</div>,
          <button
            className="btn btn-primary col-4 SearchButton"
            type="submit"
>
            Szukaj
</button>,
        ],
        "className": "Form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="form-control "
                onBlur={undefined}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="Wpisz kod waluty"
                type="text"
                value=""
/>,
              "",
              "",
              "",
            ],
            "className": "form-group col-8 Form FormInput",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form-control ",
                "onBlur": undefined,
                "onChange": [Function],
                "onFocus": [Function],
                "onKeyDown": [Function],
                "placeholder": "Wpisz kod waluty",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "",
            "",
            "",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Szukaj",
            "className": "btn btn-primary col-4 SearchButton",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Szukaj",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
